litestar/testing/request_factory.py

```diff
@@ -209,7 +211,7 @@ def _create_request_with_data(
         user: Any = None,
         auth: Any = None,
         request_media_type: RequestEncodingType = RequestEncodingType.JSON,
-        data: dict[str, Any] | BaseModel | None = None,
+        data: dict[str, Any] | DataContainerType | None = None,  # pyright: ignore
```

Comment:
not really in the same scope of the PR but it was either removing this or supporting everything else we support. I opted for the latter, since its easy enough to achieve. This is thus a micro feature- making the `request_factory` slightly better.